### ISSUE TITLE
getActionViewItem is called too often, breaking view action items that rely on not being instantiated more than once

### ISSUE THREAD

**Issue by alexr00** [microsoft/vscode#112750]
This breaks the Remote Explorer's drop down menu.

When the Remote Explorer's drop down menu is instantiated, it sets it's default value to match the current remote connection. Now, with `getActionViewItem` being called so much more often, when you try to use the drop down while connected to a remote, the default value is reset, breaking it.

You can easily repro by connecting to a remote, then trying to use the drop down in the Remote Explorer

I traced it back to this commit with git bisect: https://github.com/microsoft/vscode/commit/b44750d551debee69710fd3e641e735f69d0b61d

**Comment by alexr00** [comment]
FYI @sbatten too in case you can look at it first.

**Comment by sbatten** [comment]
looked into this. the issue is indeed bad, but I think the call is correct. when the remote selector changes, the top-level actions could hypothetically change given the new context. the remote explorer does not do this, but the new model is more unified. I think the proper fix here is that the remote explorer caches its instance of the `SwitchRemoteViewItem` since it controls instantiation.

**Comment by alexr00** [comment]
@sbatten thanks for taking a look! Unfortunately, with this change now the drop down just doesn't show sometimes, and when it does show, the styling is incorrect. I'll revert it, and we can try again.

**Comment by alexr00** [comment]
I pushed a change that fixes the issue, but it isn't as nice as the previous behavior. Leaving open to see if there's a better way.

**Comment by sandy081** [comment]
Sorry for breaking this but this uncovered an issue in the action item created by the remote explorer. An action item shall always take current state into the consideration when asked to render. I think the current fix is the right approach. You can also think of passing the item to select through action item constructor options.

`getActionItem` being called multiple times is expected and designed with `Menu` and `ActionBar`. Actions are replaced and updated in the ActionBar whenever the Menu changes which asks for `ActionItem` for each action. Hence, it is not expected to assume `getActionItem` is called only once.

So assigned the issue to @alexr00 as this needs fix in `SwitchRemoteViewItem`. Feel free to close this if it is fixed by your latest fix.

**Comment by alexr00** [comment]
In that case, my change has fixed it.

**Comment by eamodio** [comment]
@alexr00 Is this related to this?
![flash](https://user-images.githubusercontent.com/641685/106200772-7ff15880-6185-11eb-859c-15136b02fa58.gif)

**Comment by sandy081** [comment]
@eamodio It seems hover issue on tree item and is not related to this issue. Please file a separate issue for that.
